File: app/src/main/java/com/lapptelier/test/TestViewHolder.java
package com.lapptelier.test;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.lapptelier.smartrecyclerview.MyViewHolder;
import com.lapptelier.smartrecyclerview.ViewHolderInteractionListener;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author L'Apptelier SARL
 * @date 14/09/2017
 */
class TestViewHolder extends RecyclerView.ViewHolder implements MyViewHolder<String> {
    @BindView(R.id.text)
    TextView textView;

    String cellText;

    public TestViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);

    }

    @Override
    public void setItem(final String text, @NonNull final ViewHolderInteractionListener listener) {
        cellText = text;
        textView.setText(text);

        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onItemClicked(text);
            }
        });
        textView.setLongClickable(true);
    }
}

File: app/src/main/java/com/lapptelier/test/TestViewHolder.kt
package com.lapptelier.test

import androidx.recyclerview.widget.RecyclerView
import com.lapptelier.smartrecyclerview.SmartBindingViewHolder
import com.lapptelier.smartrecyclerview.ViewHolderInteraction
import com.lapptelier.smartrecyclerview.ViewHolderInteractionListener
import com.lapptelier.test.databinding.CellTestBinding

/**
 * @author L'Apptelier SARL
 * @date 14/09/2017
 */
internal class TestViewHolder(private val binding: CellTestBinding) :
    SmartBindingViewHolder<String?>(binding.root) {

    override fun setItem(
        item: String?,
        listener: ViewHolderInteractionListener,
        recyclerViewPool: RecyclerView.RecycledViewPool?
    ) {
        binding.text = item
        item?.let {
            binding.root.setOnClickListener {
                listener.onItemAction(
                    item,
                    binding.root.id,
                    ViewHolderInteraction.TAP
                )
            }
            binding.root.setOnLongClickListener {
                listener.onItemAction(
                    item,
                    binding.root.id,
                    ViewHolderInteraction.LONG_TAP
                )
                false
            }

        }
        binding.executePendingBindings()
    }
}